scripts/host-build-all.sh

```diff
@@ -5,17 +5,59 @@
 # fail on any errors
 set -e
 
-cd tools
-cd testbench
+print_usage()
+{
+    cat <<EOFUSAGE
+usage: $0 [-f]
+       -f Build testbench with compiler provided by fuzzer
+          (default path: /home/sof/work/AFL/afl-gcc)
+EOFUSAGE
+}
 
-rm -rf build_testbench
+make_testbench()
+{
+    cd "$BUILD_TESTBENCH_DIR"
 
-mkdir build_testbench
-cd build_testbench
+    rm -rf build_testbench
 
-cmake -DCMAKE_INSTALL_PREFIX=install \
-	-DCMAKE_VERBOSE_MAKEFILE=ON \
-	..
+    mkdir build_testbench
+    cd build_testbench
 
-make -j$(nproc --all)
-make install
+    cmake -DCMAKE_INSTALL_PREFIX=install \
+	  -DCMAKE_VERBOSE_MAKEFILE=ON \
+	  ..
+
+    make -j$(nproc --all)
+    make install
+}
+
+build_with_afl()
+{
+    export CC=/home/sof/work/AFL/afl-gcc
```

Comment:
We should pass this in or define the default location relative to $HOME as a global variable

Reply:
I think using $HOME will be a problem. If user tries to run this script as a root, then the $HOME would be different, which would lead to errors. It may be better to give the complete path instead of using $HOME.

Reply:
how is afl installed, can we use something like pkg-config to look it up?

Reply:
We have to build it from the sources. I don't think we have a package for it.

Reply:
>  If user tries to run this script as a root, 

I can't imagine why anyone would run this script as root. If they run _this_ as root then they probably do everything as root, including... building and installing AFL?

Now don't get me wrong: I'm not a fan of `$HOME/apl/afl/whatever` either but I think it's much better than `/home/sof/apl/afl/whatever`.The former can be created as a symbolic link that doesn't require... root access, not the latter.

Some IMHO essential and missing features are the ability to override the default and a tiny bit of logging. It could go like this:

```shell
main()
{
  : ${SOF_AFL:=$HOME/apl/afl/whatever}

export_CC_is_afl()
{
    printf 'export CC=%s\n' "${SOF_APL_AFL}"
    export CC=${SOF_APL_AFL}
}
```


```

Reply:
@marc-hb "I just kept adding sudo and it just kept working 😄 "

Reply:
... but don't worry everything's super secure because we're fuzzying now ;-)

Reply:
Updated the commit to use `$HOME` instead of `\home\` :)